### ISSUE TITLE
date()

### ISSUE THREAD

**Issue by sashsvamir** [yiisoft/yii2#16065]
### What steps will reproduce the problem?
1. Add follow code at header of `framework/yii`:
```php
echo date_default_timezone_get();
echo date('Y-m-d H:i:s', time());
// ...
```

2. Create action (to any controller):
```php
public function actionDate() {
    echo date_default_timezone_get();
    echo date('Y-m-d H:i:s', time());
}
```

3. Run above action

### What is the expected result?
```
Europe/Moscow
2018-04-09 08:18:40
Europe/Moscow
2018-04-09 08:18:40
```

### What do you get instead?
```
Europe/Moscow
2018-04-09 08:18:40
UTC
2018-04-09 05:18:40
```

### Additional info

Also i try run `php -r 'phpinfo();` where i get:
```
Europe/Moscow
2018-04-09 08:18:40
```

Note: both yii app and terminal executed from same php and using the same `php.ini` config file, there no diffs at php configs.

Also i'm compared `phpinfo()` output from yii app and from terminal command `php -r 'phpinfo();'`, there found that yii app use default timezone as `UTC`, but php running from terminal using `Europe/Moscow`. There full diff of `phpinfo()`:
```diff
--- Untitled	Mon Apr  9 07:29:42 2018
+++ Untitled2	Mon Apr  9 07:29:42 2018
@@ -209,7 +209,7 @@
 timelib version => 2017.05beta9
 "Olson" Timezone Database Version => 0.system
 Timezone Database => internal
-Default timezone => UTC
+Default timezone => Europe/Moscow
 
 Directive => Local Value => Master Value
 date.default_latitude => 31.7667 => 31.7667
@@ -962,7 +962,7 @@
 LESSOPEN => | /usr/bin/lesspipe %s
 J2REDIR => /usr/lib/jvm/java-8-oracle/jre
 LESSCLOSE => /usr/bin/lesspipe %s %s
-_ => ./yii
+_ => /usr/bin/php
 OLDPWD => /home/sash
 
 PHP Variables
@@ -994,22 +994,21 @@
 $_SERVER['LESSOPEN'] => | /usr/bin/lesspipe %s
 $_SERVER['J2REDIR'] => /usr/lib/jvm/java-8-oracle/jre
 $_SERVER['LESSCLOSE'] => /usr/bin/lesspipe %s %s
-$_SERVER['_'] => ./yii
+$_SERVER['_'] => /usr/bin/php
 $_SERVER['OLDPWD'] => /home/sash
-$_SERVER['PHP_SELF'] => ./yii
-$_SERVER['SCRIPT_NAME'] => ./yii
-$_SERVER['SCRIPT_FILENAME'] => ./yii
-$_SERVER['PATH_TRANSLATED'] => ./yii
+$_SERVER['PHP_SELF'] => Standard input code
+$_SERVER['SCRIPT_NAME'] => Standard input code
+$_SERVER['SCRIPT_FILENAME'] =>
+$_SERVER['PATH_TRANSLATED'] =>
 $_SERVER['DOCUMENT_ROOT'] =>
-$_SERVER['REQUEST_TIME_FLOAT'] => 1523247637.4257
-$_SERVER['REQUEST_TIME'] => 1523247637
+$_SERVER['REQUEST_TIME_FLOAT'] => 1523247979.8361
+$_SERVER['REQUEST_TIME'] => 1523247979
 $_SERVER['argv'] => Array
 (
-    [0] => ./yii
-    [1] => user/frontend/date
+    [0] => Standard input code
 )
 
-$_SERVER['argc'] => 2
+$_SERVER['argc'] => 1
 
 PHP License
 This program is free software; you can redistribute it and/or modify
```

I'm not sure that this not unwanted behaviour or this problem not from php, if so, please, sorry me for that issue. But now i can't find point where default timezone turns happens.

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.15.1
| PHP version      | 7.2
| Operating system |Ubuntu 16.04

**Comment by cebe** [comment]
have you configured timezone in yii applicatino config?

**Comment by sashsvamir** [comment]
Why not get In Application.php timezone from `date_default_timezone_get()`?
```diff
if (isset($config['timeZone'])) {
    $this->setTimeZone($config['timeZone']);
    unset($config['timeZone']);
- } elseif (!ini_get('date.timezone')) {
-    $this->setTimeZone('UTC');
+ } else {
+     $this->setTimeZone(date_default_timezone_get());
}
```

I think `date_default_timezone_get()` function return is more expected result. Because `date()` (used at writing time of log events), by running at command line return system date, not UTC.

**Comment by cebe** [comment]
this avoids PHP warning if timezone is not configured in php config.

https://github.com/yiisoft/yii2/blob/e6f5c46cdc64e52364d868f064e3548338e929d7/framework/base/Application.php#L244-L249

> https://secure.php.net/manual/de/function.date-default-timezone-get.php
> A warning is shown when this stage is reached. Do not rely on it to be guessed correctly, and set date.timezone to the correct timezone instead.

**Comment by sashsvamir** [comment]
Now i'm check `date_default_timezone_get()` on php 7.1 and 7.2 with `error_reporting=-1` at config and there no warnings. Also at [php.net](http://php.net/manual/ru/function.date-default-timezone-get.php) saying, that in case only in php 5.4 uses warning. Because yii 2.1 require php => 7.1 may be there using `date_default_timezone_get` instead getting timezone from php.ini

**Comment by cebe** [comment]
seems PHP 7 changed this behavior: https://3v4l.org/YhWHq
should be adjusted in 2.1.

**Comment by cebe** [comment]
there was an RFC, implemented in php 7.0: https://wiki.php.net/rfc/date.timezone_warning_removal https://github.com/php/php-src/pull/1029

**Comment by sashsvamir** [comment]
so can i suggest pr like this: #16070

**Comment by cebe** [comment]
fixed by #16070
